ui/src/wallet.ts

```diff
@@ -0,0 +1,135 @@
+import WalletConnect from "@walletconnect/client";
+import { convertUtf8ToHex } from "@walletconnect/utils";
+import QRCodeModal from "@walletconnect/qrcode-modal";
+import WalletConnectWeb3Provider from "@walletconnect/web3-provider";
+import Web3 from "web3";
+import { writable, Readable } from "svelte/store";
+
+export enum Status {
+  Connected = "CONNECTED",
+  NotConnected = "NOT_CONNECTED",
+}
+
+type State =
+  | { status: Status.NotConnected }
+  | { status: Status.Connected; connected: Connected };
+
+interface Connected {
+  chainId: number;
+  account: {
+    address: string | null;
+    balance: string | null;
+  };
+}
+
+interface Wallet extends Readable<State> {
+  connect(): Promise<void>;
+  disconnect(): Promise<void>;
+  testSign(): Promise<void>;
+}
+
+export function makeWallet(): Wallet {
+  const stateStore = writable<State>({ status: Status.NotConnected });
+  const connector = new WalletConnect({
+    bridge: "https://bridge.walletconnect.org",
+    qrcodeModal: QRCodeModal,
+  });
+  const provider = new WalletConnectWeb3Provider({
+    infuraId: "47a262b47b864e69ab21425bacf4c3df",
+    // Comment this below to connect to the mainnet instead
+    rpc: {
+      1: "http://localhost:8545",
+    },
+    connector,
+  });
+  const web3 = new Web3(provider as any);
+
+  // Connect to a wallet using walletconnect
+  async function connect() {
+    // TODO handle error
+    const accountAddresses: string[] = (await provider.enable()) as any;
```

Comment:
Why is this an `any`?

Reply:
this had been written by Thomas. `provider.enable()` returns `Promise<unknown>`, I assume the `as any` was the quickest workout. Changed it to `as string[]`.